Automattic/Gravatar-SDK-iOS
# Update openapi spec

### Description

This PR updates the open api spec.

Some changes to note:

- Fetch Identity is not longer required, so it has been removed.
- Avatars request now returns the selected avatar. This replaces the identity fetch.
- Image upload error model for status 400 (i.E: `uncropped_image` and `unsupported_image`).
  - The UI has been updated to not prompt a retry in these cases. A cross will be shown which will remove the failed image when tapped. 
- Avatar selection endpoint has been updated to the latest spec. 

![CleanShot 2024-09-20 at 17 49 50](https://github.com/user-attachments/assets/aa55a4ca-65a7-4443-991c-9dd1e9eea580)


*Note*: I don't particularlt like that all the openapi structs are public. Many of these we use them internally only.

### Testing Steps

Test all avatar picker flows:
- Loading avatars.
- Selecting different avatar.
- Upload avatar.

Reproduce upload avatar error:
- This will reproduce an image-not-square error
  - Go to `AvatarPickerViewModel.swift:248`
    - Comment the `guard` statement.  
  - Go to `AvatarPickerViewModel.swift:261`
    - Change one of the sizes (adding an extra number, for example:
      -  `let squareSize = CGSize(width: squareSide, height: squareSide + 10)`
  -  Build and run
  - Try to upload an image.
    - Check that the error appear and there's no re-upload option.
    - Tap on the X on the avatar, the avatar should disappear.
- Check a different error as uploading avatar without internet connection.
  - Check that re-upload option is present.
